src/main/java/com/gradle/exportapi/EventProcessor.java

```diff
@@ -218,10 +218,17 @@ public static void persist(EventProcessor processor) {
         Build currentBuild = processor.currentBuild;
         currentBuild.resolveStatus();
 
-        currentBuild.setId( insertBuild(currentBuild) );
+        Optional<Long> buildTableId = BuildDAO.getBuildTableId(currentBuild);
+        if (buildTableId.isPresent()) {
+            log.warn(String.format("Skipped build %s that is already in db. This could entail an incomplete export from previous run.",
```

Comment:
No reason to use String.format

```java
log.warn("Skipping build {} as it is already in the database.", currentBuild.getBuildId());
```
Also, remove the warning about an incomplete export.  That's not relevant here.